### PR TITLE
Added homework 8

### PR DESCRIPTION
**Pull request by ThinamXx**

### CODE REVIEW

File: course-zoomcamp/cohorts/2022/08-deep-learning/homework.md [hunk 1]
```diff
@@ -0,0 +1,159 @@
+## Homework
+
+### Dataset
+
+In this homework, we'll build a model for predicting if we have an image of a dino or a dragon. For this, we will use the "Dino or Dragon?" dataset that can be downloaded from [Kaggle](https://www.kaggle.com/datasets/agrigorev/dino-or-dragon). 
+
+In the lectures we saw how to use a pre-trained neural network. In the homework, we'll train a much smaller model from scratch. 
+
+**Note:** You don't need a computer with a GPU for this homework. A laptop or any personal computer should be sufficient.
```
Comment (alexeygrigorev) [inline course-zoomcamp/cohorts/2022/08-deep-learning/homework.md:9]: We'll be able to get some GPU compute from saturn cloud. I'll add details later

File: course-zoomcamp/cohorts/2022/08-deep-learning/homework.md [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+**Note:** You don't need a computer with a GPU for this homework. A laptop or any personal computer should be sufficient. 
+
+
+### Data Preparation
+
+The dataset contains around 1900 images of dinos and around 1900 images of dragons. 
+
+Now we need to split this data into train and validation
+
+* Create a `train` and `test` folders
```
Comment (alexeygrigorev) [inline course-zoomcamp/cohorts/2022/08-deep-learning/homework.md]: The data is already split into folders actually, so no need to do it

File: course-zoomcamp/cohorts/2022/08-deep-learning/homework.md [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+    train_generator,
+    epochs=10,
+    validation_data=validation_generator
+)
+```
+
+
+### Question 3
+
+What is the median of training accuracy for this model?
```
Comment (alexeygrigorev) [inline course-zoomcamp/cohorts/2022/08-deep-learning/homework.md]: Maybe we can be a bit more explicit of what we're asking? I suppose it's the training accuracy for all the epochs?

File: course-zoomcamp/cohorts/2022/08-deep-learning/homework.md [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+
+- 0.40
+- 0.60
+- 0.90
+
+### Question 4
+
+What is the standard deviation of training loss for this model?
```
Comment (alexeygrigorev) [inline course-zoomcamp/cohorts/2022/08-deep-learning/homework.md]: Same here

File: course-zoomcamp/cohorts/2022/08-deep-learning/homework.md [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+- 0.37
+
+### Question 6
+
+What's the average of validation accuracy for the last 5 epochs (from 6 to 10)
+for the model trained with augmentations?
+
+- 0.84
+- 0.54
+- 0.44
```
Comment (alexeygrigorev) [inline course-zoomcamp/cohorts/2022/08-deep-learning/homework.md:142]: Can we add an 4th option for all the questions?

File: course-zoomcamp/cohorts/2022/08-deep-learning/homework.md [hunk 6]
```diff
@@ -0,0 +1,157 @@
...
+### Data Preparation
+
+The dataset contains around 1900 images of dinos and around 1900 images of dragons. 
+
+The dataset contains separate folders for training and validation. 
+
+
+### Model
+
+For this homework we will use Convolutional Neural Network (CNN. Like in the lectures, we'll use Keras.
```
Comment (SVizor42) [inline course-zoomcamp/cohorts/2022/08-deep-learning/homework.md]: Сlosing bracket

File: course-zoomcamp/cohorts/2022/08-deep-learning/homework.md [hunk 7]
```diff
@@ -0,0 +1,157 @@
...
+
+You need to develop the model with following structure:
+
+* The shape for input should be `(150, 150, 3)`
+* Next, create a covolutional layer ([`Conv2D`](https://keras.io/api/layers/convolution_layers/convolution2d/)):
```
Comment (SVizor42) [inline course-zoomcamp/cohorts/2022/08-deep-learning/homework.md]: convolutional

File: course-zoomcamp/cohorts/2022/08-deep-learning/homework.md [hunk 8]
```diff
@@ -0,0 +1,157 @@
...
+For the next two questions, use the following data generator for both train and validation:
+
+```python
+ImageDataGenerator(rescale=1./255)
+```
+
+* We don't need to do any additional pre-processing for the images.
+* When reading the data from train/val directories, check the `class_mode` parameter. Which value should it be for a binary classification problem?
+* Use `batch_size=20`
+* Use `shuffle=True` for both training and validaition
```
Comment (SVizor42) [inline course-zoomcamp/cohorts/2022/08-deep-learning/homework.md]: validation

**Comment by alexeygrigorev** [comment]
Thank you!
